Tryouts/Plugins/ApplicationPlugins/AnotherWpfApp/ViewModel.cs

```diff
@@ -0,0 +1,64 @@
+﻿using MorganStanley.ComposeUI.Tryouts.Core.Abstractions;
+using MorganStanley.ComposeUI.Tryouts.Core.Utilities;
+using Subscriptions;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace AnotherWpfApp
+{
+    public class ViewModel : ViewModelBase
+    {
+        CancellationTokenSource? _cts = new CancellationTokenSource();
+
+        private readonly ISubscriptionClient _subscriptionClient;
+
+        IDisposable? _subscription;
+        public ViewModel()
+        {
+            _subscriptionClient = ((App)App.Current).Container.Resolve<ISubscriptionClient>();
```

Comment:
I understand that this is just a 'tryout' but I wouldn't use a publicly exposed container instance to resolve dependencies. I believe dependencies should be visible from the outside (e.g.: ctor arguments), and the container should only be accessed directly when it's being configured (composition root).

Reply:
great point! ViewModel should not be aware of the container. I made container private in all cases and did the bootstrapping where it belongs - in App.xaml.cs OnStartup() method.